src/mongocrypt-marking.c

```diff
@@ -457,13 +457,39 @@ _mongocrypt_fle2_placeholder_to_insert_update_ciphertext (
 
    _mongocrypt_buffer_from_iter (&value, &placeholder->v_iter);
 
-   if (!_mongocrypt_fle2_placeholder_common (kb,
-                                             &common,
-                                             &placeholder->index_key_id,
-                                             &value,
-                                             true, /* derive tokens using counter */
-                                             placeholder->maxContentionCounter,
-                                             status)) {
+   int64_t contentionFactor = 0;
+   if (placeholder->maxContentionCounter < 0) {
+      CLIENT_ERR ("maxContentionCounter must be non-negative, got: %" PRId64,
+                  placeholder->maxContentionCounter);
+      goto fail;
+   }
+   if (placeholder->maxContentionCounter > 0) {
+      if (placeholder->maxContentionCounter < 0) {
```

Comment:
How is this check expected to pass?

It was already checked on line 461, and it's antithetical to the check on line 466